### ISSUE TITLE
Duplicate products on shopping list when adding from meal plan

### ISSUE THREAD

**Issue by DeeeeLAN** [grocy/grocy#402]
If I have two meals on the meal plan that both use an ingredient that I need, when I add all ingredients to the shopping list, that ingredient gets entered twice. It would be nice if it only entered once and increased the quantity on the second addition. I guess this might be slightly tricky for weird quantity units? 

It would also be nice if it maintained what recipe(s) the item originated from instead of just the weekly meal plan. I tried adding the recipes individually but then the items only get added to the list once, which is bad because then I won’t buy enough. 

![68F76E9C-0329-4A92-978E-9268ABAE6C39](https://user-images.githubusercontent.com/12877032/65896137-84bfb580-e361-11e9-857c-c9675c52be60.jpeg)

**Comment by CRTified** [comment]
> I tried adding the recipes individually but then the items only get added to the list once, which is bad because then I won’t buy enough.

You can prevent this by enabling the checkbox for the recipe to disable checking the shopping list.
But this still prevents the Items from being merged, which causes pretty complex shopping lists on my side.

**Comment by CRTified** [comment]
My makeshift solution is now the following:

1. Create a second shopping list
2. Add all your stuff to the first shopping list
3. Execute this SQL statement on the DB:

```sql
DELETE FROM shopping_list WHERE shopping_list_id = 2;
INSERT INTO shopping_list ("product_id", note, amount, row_created_timestamp, shopping_list_id, done) 
  SELECT product_id, GROUP_CONCAT(note, "; "), SUM(amount), row_created_timestamp, 2 as shopping_list_id, done 
  FROM "shopping_list" WHERE shopping_list_id = 1 GROUP BY "product_id";

```

**Comment by berrnd** [comment]
Duplicate of / merged into #840.
